src/Features/Core/Portable/ImplementInterface/AbstractImplementInterfaceService.AccessibilityHelper.cs

```diff
@@ -0,0 +1,114 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.ImplementInterface
+{
+    internal abstract partial class AbstractImplementInterfaceService
+    {
+        private static class AccessibilityHelper
+        {
+            public static bool IsLessAccessibleThan(ISymbol first, INamedTypeSymbol second)
+            {
+                if (first.DeclaredAccessibility <= Accessibility.NotApplicable ||
+                    second.DeclaredAccessibility <= Accessibility.NotApplicable)
+                {
+                    return false;
+                }
+
+                if (first.DeclaredAccessibility < second.DeclaredAccessibility)
+                {
+                    return true;
+                }
+
+                switch (first)
+                {
+                    case IPropertySymbol propertySymbol:
+                        if (IsTypeLessAccessibleThanOtherType(propertySymbol.Type, second))
+                        {
+                            return true;
+                        }
+
+                        if (propertySymbol.GetMethod is not null)
+                        {
+                            foreach (var parameter in propertySymbol.GetMethod.Parameters)
```

Comment:
instead of walking the parameters here, just recurs on the .GetMethod and .SetMethod.